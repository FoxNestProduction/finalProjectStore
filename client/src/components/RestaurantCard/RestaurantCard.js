import React, { useState } from 'react';
import Box from '@mui/material/Box';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import PropTypes from 'prop-types';
import { Stack } from '@mui/material';
import Button from '@mui/material/Button';
import BookmarkIcon from '../../assets/svgComponents/BookmarkIcon';
import RatingItem from '../Rating/Rating';

const RestaurantCard = ({ rating, name, imageUrl, description, styleWidth }) => {
  const [desc, setDesc] = useState(false);
  console.log(description);
  return (
    <Card
      sx={{
        display: 'flex',
        alignItems: 'center',
        flexWrap: 'wrap',
        width: '100%',
        maxWidth: { ...styleWidth },
        margin: '0 auto',
        marginBottom: '50px',
        boxShadow: '6px 71px 35px 0px rgba(229, 229, 229, 0.70)',
        '&:hover .MuiCardMedia-img': {
          transform: 'scale(1.05)',
        },
      }}
    >
      <Box
        sx={{
          overflow: 'hidden',
          width: '100%',
          height: '100%',
        }}
      >
        <CardMedia
          component="img"
          image={imageUrl}
          alt={name}
          sx={{
            transition: '.3s',
            height: { mobile: '140px', tablet: '220px', lgTablet: '268px', desktop: '378px' },
          }}
        />
      </Box>
      <Stack
        direction="row"
        width="100%"
        flexWrap="wrap"
        justifyContent="space-between"
        alignItems="center"
        sx={{ margin: '25px' }}
      >
        <CardHeader
          title={(
            <Typography
              variant="h3"
              color="text.primary"
              sx={{
                fontSize: {
                  mobile: '1.2rem',
                  tablet: '1.3rem',
                  desktop: '1.5rem',
                },
              }}
            >
              {name}
            </Typography>
        )}
          sx={{
            padding: '0',
            marginBottom: {
              desktop: '10px',
            },
            width: { mobile: '100%', lgTablet: '40%' },
          }}

        />
        <CardContent
          sx={{
            display: 'flex',
            alignItems: 'center',
            gap: '10px',
            padding: '0',
            width: { mobile: '87%', lgTablet: '55%' },
            flexWrap: 'noWrap',
          }}
        >
          <RatingItem ratingValue={rating} />

          <Typography
            variant="body1"
            color="text.secondary"
            sx={{
              width: {
                desktop: 67,
                tablet: '100%',
              },
            }}
          >
            24min â€¢
          </Typography>

        </CardContent>

        <CardActions
          disableSpacing
          sx={{
            padding: '0',
            paddingRight: '18px',
            width: { mobile: '13%', lgTablet: '5%' },
          }}
        >
          <IconButton
            aria-label="add to favorites"
            sx={{
              padding: '0',
              width: {
                mobile: 35,
                tablet: 41,
              },
              height: {
                mobile: 35,
                tablet: 41,
              },
            }}
          >
            <BookmarkIcon />
          </IconButton>
        </CardActions>
      </Stack>
      <Stack direction="row" justifyContent="flex-end" width="100%" m="15px">
        <Button
          disableSpacing
          variant="contained"
          sx={{
            borderRadius: '14px',
            border: '1px solid primary.main',
            fontSize: { mobile: '14px', lgTablet: '18px' },
            transition: 'background-color 0.3s ease, box-shadow 0.3s ease, color 0.3s ease',
            ':hover': {
              backgroundColor: 'primary.hover',
              boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.3)',
              zIndex: 100,
            },
            '&:active': {
              backgroundColor: 'common.white',
              boxShadow: '0px 4px 6px rgba(0, 0, 0, 0.3)',
              color: 'primary.main',
              border: '1px solid',
            },
          }}
          onClick={(e) => {
            e.preventDefault();
            setDesc((prev) => !prev);
          }}
        >
          {desc ? 'Close' : 'Show more'}
        </Button>
      </Stack>
      <Typography
        variant="description"
        component="p"
        margin={desc ? '25px' : '0'}
        sx={{ textAlign: 'justify' }}
      >
        {desc && description}
      </Typography>
    </Card>
  );
};

RestaurantCard.propTypes = {
  rating: PropTypes.number,
  name: PropTypes.string,
  imageUrl: PropTypes.string,
  description: PropTypes.bool,
  styleWidth: PropTypes.object,
};

RestaurantCard.defaultProps = {
  rating: 3,
  name: 'The Chicken King',
  imageUrl: './img/restaurants/Resturent01.jpg',
  description: false,
  styleWidth: {},
};

export default RestaurantCard;
