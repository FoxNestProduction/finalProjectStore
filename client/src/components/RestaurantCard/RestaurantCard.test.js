import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import RestaurantCard from './RestaurantCard';

describe('RestaurantCard', () => {
  test('renders RestaurantCard component', () => {
    const props = {
      rating: 4,
      name: 'Test Restaurant',
      imageUrl: 'https://https://osvitanova.com.ua/ckeditor_assets/pictures/6254/content_1.jpg',
      description: '123',
      styleWidth: { mobile: '100%', lgTablet: '50%' },
    };

    const { asFragment } = render(
      <Router>
        <RestaurantCard {...props} />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Test Restaurant')).toBeInTheDocument();
    expect(screen.getByText('24min â€¢')).toBeInTheDocument();
    expect(screen.getByAltText('Test Restaurant')).toHaveAttribute('src', 'https://https://osvitanova.com.ua/ckeditor_assets/pictures/6254/content_1.jpg');

    expect(screen.getByText('Show more')).toBeInTheDocument();
    fireEvent.click(screen.getByText('Show more'));

    expect(screen.getByText('123')).toBeInTheDocument();

    expect(screen.getByText('Close')).toBeInTheDocument();
  });
});
