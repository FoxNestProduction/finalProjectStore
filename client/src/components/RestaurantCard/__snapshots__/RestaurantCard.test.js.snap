// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard renders RestaurantCard component 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-j929nk-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-gpemya"
    >
      <img
        alt="Test Restaurant"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-fh02g9-MuiCardMedia-root"
        src="https://https://osvitanova.com.ua/ckeditor_assets/pictures/6254/content_1.jpg"
      />
    </div>
    <div
      class="MuiStack-root css-rwxpvi-MuiStack-root"
    >
      <div
        class="MuiCardHeader-root css-4mdd9q-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-xz6qnp-MuiTypography-root"
          >
            Test Restaurant
          </h3>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-oq62jt-MuiCardContent-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <span
            aria-label="4 Stars"
            class="MuiRating-root MuiRating-sizeSmall Mui-readOnly MuiRating-readOnly css-1nmdjm5-MuiRating-root"
            role="img"
          >
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="StarRateRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="StarRateRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="StarRateRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                  data-testid="StarRateRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="StarRateRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-w9h7d4-MuiTypography-root"
        >
          24min â€¢
        </p>
      </div>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-mkkqzv-MuiCardActions-root"
      >
        <button
          aria-label="add to favorites"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1n91ue6-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            fill="none"
            height="41"
            viewBox="0 0 41 41"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="20.3574"
              cy="20.6406"
              fill="#DBD9EE"
              r="20"
            />
            <path
              d="M24.7408 11.2928C21.8454 10.8074 18.8738 10.8074 15.9784 11.2928C14.9206 11.4701 14.0782 12.2822 13.8661 13.3351C12.8944 18.1588 12.8414 23.1226 13.7098 27.966L14.0634 29.9379C14.1724 30.5459 14.9136 30.7889 15.3615 30.3635L19.6291 26.3093C20.0384 25.9204 20.6807 25.9204 21.0901 26.3093L25.3576 30.3635C25.8055 30.7889 26.5468 30.5459 26.6558 29.9379L27.0093 27.966C27.8777 23.1226 27.8247 18.1588 26.8531 13.3351C26.641 12.2822 25.7986 11.4701 24.7408 11.2928Z"
              fill="#6C5FBC"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiStack-root css-iaetsx-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-5jihz5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Show more
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <p
      class="MuiTypography-root MuiTypography-description css-9psp93-MuiTypography-root"
    />
  </div>
</DocumentFragment>
`;
